book_source/06_reference/04_docker/05_building_images.Rmd

```diff
@@ -0,0 +1,141 @@
+
+First, in a terminal window, navigate (`cd`) into the PEcAn source code root directory.
+From there, the general syntax for building an image looks like the following:
+
+```bash
+docker build -t pecan/<image name>:<image version> -f docker/base/Dockerfile.<image name> .
+```
+
+For instance, to build a local version of the `pecan/depends:latest` image, you would run:
+
+```bash
+docker build -t pecan/depends:latest -f docker/base/Dockerfile.depends .
+```
+
+The breakdown of this command is as follows:
+
+- `docker build` -- This is the core command.
+The standard syntax is `docker build [OPTIONS] <PATH>`, where `<PATH>` refers to the directory to be used as the "build context".
+The "build context" is the working directory assumed by the Dockerfiles.
+In PEcAn, this is always the PEcAn source code root directory, which allows Dockerfiles to use instructions such as `COPY web/workflow.R /work/`.
+In this example, the `<PATH>` is set to the current working directory, i.e. `.` because we are already in the PEcAn root directory.
+If you were located in a different directory, you would have to provide a path to the PEcAn source code root directory.
+Also, by default, `docker build` will look for a Dockerfile located at `<PATH>/Dockerfile`, but this is modified by the `-f` option described below.
+
+- `-t pecan/depends:latest` -- The `-t/--tag` option specifies how the image will be labeled.
+By default, Docker only defines unique image IDs, which are hexidecimal strings that are unintuitive and hard to remember.
+Tags are useful for referring to specific images in a human-readable way.
+Note that the same unique image can have multiple tags associated with it, so it is possible for, e.g. `pecan/depends:latest`, `pecan/depends:custom`, and even `mypecan/somethingelse:20.0` to refer to the same exact image.
+To see a table of all local images, including their tags and IDs, run `docker image ls`.
+	- **NOTE**: PEcAn's `docker-compose.yml` can be configured via the `PECAN` environment variable to point at different versions of PEcAn images.
+	By default, it points to the `:latest` versions of all images.
+	However, if you wanted to, for instance, build `:local` images corresponding to your local source code and then run that version of PEcAn, you would run:
+	
+	```
+	PECAN=local docker-compose -p pecan up -d
+	```
+	
+	This is an effective way to do local development and testing of different PEcAn versions, as described [below](#docker-local-devel).
+
+- `-f docker/base/Dockerfile.depends` -- The `-f/--file` tag is used to provide an alternative location and file name for the Dockerfile.
+The convention in PEcAn is to put Dockerfiles for core PEcAn functionality in `docker/base/` and for specific models in `docker/models/`, and to name these files `Dockerfile.<image name>`.
+
+#### Local development and testing with Docker {#docker-local-devel}
```

Comment:
is there / should there be a separate section on 'deploying a production PEcAn instance using docker'?

Reply:
At some point, sure. It shouldn't be much (if at all) different from the instructions in the Quickstart. But, any differences from the standard "Quickstart" workflow are not something I've worked out yet, so I'm going to punt on it until we have something to document.